packages/react-flight-dom-webpack/src/ReactFlightClientWebpackBundlerConfig.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+export type ModuleMetaData = {
+  id: string,
+  chunks: Array<string>,
+};
+
+type Thenable = {
+  then(resolve: () => mixed, reject: (mixed) => mixed): mixed,
+  ...
+};
+
+// The chunk cache contains all the chunks we've preloaded so far.
+// If they're still pending they're a thenable. This map also exists
+// in Webpack but unfortunately it's not exposed so we have to
+// replicate it in user space. null means that it has already loaded.
+const chunkCache: Map<string, null | Thenable> = new Map();
+
+// Returning null means that all dependencies are fulfilled and we
+// can synchronously require the module now. A thenable is returned
+// that when resolved, means we can try again.
+export function preloadModule(moduleData: ModuleMetaData): null | Thenable {
+  let moduleEntry = require.cache[moduleData.id];
+  if (moduleEntry) {
+    // Fast exit if this module has already been loaded.
+    return null;
+  }
+  let chunks = moduleData.chunks;
+  let anyRemainingThenable = null;
+  for (let i = 0; i < chunks.length; i++) {
```

Comment:
This code snippet doesn't work for more than 1 chunks, as `anyRemainingThenable` is overwritten in each iteration.

You probably want to use `Promise.all` here?

Reply:
This will be recalled each time a new chunk resolves which will then get the next one and so on. However in practice it most likely won’t get as far as even retrying the second time until everything has been resolved.

So in practice we get one promise, then wait for it, schedule work to try again, in the meantime all of them resolve and then we try again just to confirm they all resolved. If not we’ll do the whole thing again.